### PR TITLE
chore(ruff): #80 apply preliminary rules

### PR DESCRIPTION
**Pull request by cmp0xff**
- `ignore = ["D", "FA", "PLR", "S", "PD", "SLF", "PT", "FBT", "C", "ANN", "EM", "TRY", "ARG", "FIX", "TD", "TID", "ERA", "INP", "E", "SIM", "RUF", "RET"]` in `pyproject.toml` is preliminary, such that all fixes are automatic.
- `isort` is superseded

### CODE REVIEW

File: hamilflow/models/brownian_motion.py [hunk 1]
```diff
@@ -181,7 +179,7 @@ def _trajectory(self, n_new_steps: int, seed: int) -> "npt.NDArray[np.float64]":
         )
 
         step_history = np.concatenate(
-            (np.expand_dims(self.initial_condition.x0, axis=0), step_history)
+            (np.expand_dims(self.initial_condition.x0, axis=0), step_history),
```
Comment (emptymalei) [inline hamilflow/models/brownian_motion.py:182]: Ruff adds comma to many function calls and instantiations even for single args. This is a bit strange. Is there a PEP for this style?

File: hamilflow/models/harmonic_oscillator.py [hunk 2]
```diff
@@ -277,7 +274,7 @@ def _x_critical_damped(self, t: "Sequence[float] | npt.ArrayLike") -> npt.ArrayL
         """
         t = np.array(t, copy=False)
         return self.initial_condition.x0 * np.exp(
-            -self.system.zeta * self.system.omega * t
+            -self.system.zeta * self.system.omega * t,
```
Comment (emptymalei) [inline hamilflow/models/harmonic_oscillator.py:277]: e.g. here

File: hamilflow/models/harmonic_oscillator.py [hunk 3]
```diff
@@ -349,11 +343,11 @@ def __init__(
     ) -> None:
         self.system = HarmonicOscillatorSystem.model_validate(system)
         self.initial_condition = ComplexSimpleHarmonicOscillatorIC.model_validate(
-            initial_condition
+            initial_condition,
```
Comment (emptymalei) [inline hamilflow/models/harmonic_oscillator.py:346]: e.g. or here

**Comment by emptymalei** [comment]
Is there a conflict with black? I feel it's quite aggressive. I heard similar things in the past but this is the first time I have ever seen it in action.
In general it looks good.

File: pyproject.toml [hunk 4]
```diff
@@ -65,6 +65,23 @@ build-backend = "poetry.core.masonry.api"
...
+select = ["ALL"]
+ignore = ["D", "FA", "PLR", "S", "PD", "SLF", "PT", "FBT", "C", "ANN", "EM", "TRY", "ARG", "FIX", "TD", "TID", "ERA", "INP", "E", "SIM", "RUF", "RET"]
+
+
+[tool.ruff.lint.isort]
+known-first-party = ["hamilflow"]
+
+
+[tool.black]
+line-length = 120
```
Comment (emptymalei) [inline pyproject.toml:82]: Oh this is too long.

Comment (emptymalei) [reply pyproject.toml:82]: Unless there is a very strong argument for 120, I would vote for black default line length:

https://test-black.readthedocs.io/en/latest/the_black_code_style.html#line-length

**Comment by cmp0xff** [comment]
> Is there a conflict with black? I feel it's quite aggressive. I heard similar things in the past but this is the first time I have ever seen it in action. In general it looks good.

I do not see a conflict in my experiences.

File: pyproject.toml (hunk 4 above)
Comment (cmp0xff) [reply pyproject.toml]: 21d5c96f33624883836a31f03973d391b216a67a

**Review by emptymalei — APPROVED** [review]

**Pull request merged**